### DISCUSSION TITLE
Improve display of model specific data (2 part question)

### DISCUSSION THREAD

**Discussion by ktiedt** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#273] [Ideas]
Would it be possible during main.cpp's setup process... to extract useful config values and write them to /config.json in LittleFS that way the webUI can access that data and display it when appropriate? (my C/C++ skills are very tragic, so I'm pretty sure its possible, just not sure how)

Would be a 1 time write operation each boot cycle, so if you reflashed your config with a new model # for example, a refresh of the website would render the new model info for example?

Second question.... how much space is allocatable for the web pages? I know this should be an easy thing to find but this is the first project I've ever worked on that had both a flashable filesystem and a firmware...

**Comment by visualapproach** [comment]
Yes, setup is already reading and writing config files to the file system. Can you elaborate on what things would need saving? Just the model number?
The NodeMCU 12E has 4 MB of flash. 1 MB for the sketch, 1 MB for OTA, and almost the rest for file system I think. It can be altered.

**Reply by ktiedt** [reply]
Currently it seems like knowing what model your configured for is a no brained, but really any run time things that could be useful for debugging could be useful. Off the top of my head model is the only significant config thing ATM though?

**Reply by visualapproach** [reply]
It's not saved, since it's hardcoded, but the info is transmitted to web page and over mqtt now.

**Reply by ktiedt** [reply]
That is a nice improvement, but would be nice if it wasnt reliant on an external feature.

**Reply by visualapproach** [reply]
I don't follow. Can you elaborate?

**Reply by ktiedt** [reply]
MQTT is an external optional dependency, not everyone will have it running, so they will not gain the advantage of that additional data being in MQTT.

I was simply saying it would be nice if that wasn't tied to MQTT for those users.

**Reply by visualapproach** [reply]
Oh, but it's sent over MQTT AND to the web interface.

**Reply by ktiedt** [reply]
Ohhh well then, I misunderstood the implications of your statement. Thanks for clarifying!

**Reply by visualapproach** [reply]
No worries 😉
